python/vowpalwabbit/sklearn_vw.py

```diff
@@ -508,14 +508,31 @@ class VWClassifier(SparseCoefMixin, ThresholdingLinearClassifierMixin, VW):
     note - don't try to apply link='logistic' on top of the existing functionality
     """
 
-    def __init__(self, **params):
+    def __init__(self, X=None, y=None, sample_weight=None, **params):
```

Comment:
Why is init taking the features and labels? What if you want to predict/learn with different examples?

Reply:
My initial thought was if the data is passed the initialized model could be trained as well. If its not passed, user can still call fit/predict. Do we only require object creation or should it be training+testing in the constructor requiring X_train, y_train, X_test, y_test with defaults as None?

Reply:
The constructor should not be training the model